openapi/openapi/src/main/scala/endpoints4s/openapi/model/OpenApi.scala

```diff
@@ -1467,38 +1576,49 @@ object Schema {
         description: Option[String],
         example: Option[ujson.Value],
         title: Option[String]
-    ): AllOf = new AllOf(schemas, description, example, title)
+    ): AllOf = new AllOf(schemas, description, example, title, None)
 
   }
 
   final class Reference private (
       val name: String,
       val original: Option[Schema],
-      val description: Option[String]
+      val description: Option[String],
+      val example: Option[ujson.Value],
+      val title: Option[String], // you probably want the title on the original schema!
+      override val default: Option[ujson.Value]
   ) extends Schema
       with Serializable {
 
-    override val example: None.type = None // Reference objects can’t have examples
-    override val title: None.type = None // Reference objects can’t have a title
```

Comment:
These will cause binary incompatibility due to the type going from `None.type` to `Option[String]`. Any ideas for resolving that?

Reply:
I don’t think `None` has methods that don’t exist on `Option[String]`, so I think it’s OK to add a MiMa filter here.

Reply:
Is it similarly OK to add exclusions for the `private` constructors then? For instance, the one on the lines that follow, eg:

```scala
     @deprecated("use `Reference.apply` instead", "3.1.0")
     private[openapi] def this(
         name: String,
         original: Option[Schema],
         description: Option[String]
     ) = this(name, original, description, None, None, None)
```

These should not be reachable outside of the `openapi` project since they are marked as `private`. My guess is that Mima flags them because they aren't actually `private` in the JVM class file sense, since they have to be accessible in the companion object for the `apply` method.

Reply:
Yes